lib/t/accounts.ex

```diff
@@ -425,7 +430,31 @@ defmodule T.Accounts do
   end
 
   def get_profile!(user_id) when is_binary(user_id) do
-    Repo.get!(Profile, user_id)
+    {profile, min_age, max_age, distance} =
+      Profile
+      |> where([p], p.user_id == ^user_id)
+      |> join(:left, [p], min in MinAgePreference, on: p.user_id == min.user_id)
+      |> join(:left, [p, min], max in MaxAgePreference, on: p.user_id == max.user_id)
+      |> join(:left, [p, min, max], d in DistancePreference, on: p.user_id == d.user_id)
+      |> select([p, min, max, d], {p, min.age, max.age, d.distance})
+      |> Repo.one()
+
+    gender_preference =
+      case GenderPreference
+           |> where([g], g.user_id == ^user_id)
+           |> select([g], g.gender)
+           |> Repo.all() do
+        [] -> nil
+        some -> some
+      end
+
+    %Profile{
+      profile
+      | gender_preference: gender_preference,
+        min_age: min_age,
+        max_age: max_age,
+        distance: distance
```

Comment:
need this to conveniently render OwnProfile afterwards

Reply:
you can do this in `select`